proxy.go

```diff
@@ -188,6 +216,207 @@ func (proxy *proxy) releaseToken(token Token) (*tokenInfo, error) {
 	return info, nil
 }
 
+var storeStateDir = "/var/lib/clear-containers/proxy/"
+
+type proxyStateOnDisk struct {
+	SocketPath      string   `json:"socket_path"`
+	EnableVMConsole bool     `json:"enable_vm_console"`
+	ContainerIDs    []string `json:"container_ids"`
+}
+
+// returns false if it's a clean start (i.e. no state is stored) or restoring failed
+func (proxy *proxy) restoreState() bool {
+	proxyStateFilePath := storeStateDir + "proxy_state.json"
+	if _, err := os.Stat(storeStateDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(storeStateDir, 0755); err != nil {
+			proxyLog.Errorf("Couldn't create directory %s: %v",
+				storeStateDir, err)
+		}
+		return false
+	}
+	if _, err := os.Stat(proxyStateFilePath); os.IsNotExist(err) {
+		return false
+	}
+
+	fdata, err := ioutil.ReadFile(proxyStateFilePath)
+	if err != nil {
+		proxyLog.Errorf("Couldn't recover from %s: %v", proxyStateFilePath, err)
+		return false
+	}
+
+	var proxyState proxyStateOnDisk
+	err = json.Unmarshal(fdata, &proxyState)
```

Comment:
Something to consider: what happens when we change the proxy state file format? The 3 main approaches are:

- Don't handle it and just error if the on-disk state doesn't match what the proxy expects.
- Add a version field to `vmStateOnDisk` to allow the proxy to determine how to deal with a different format state file.
- Make the code "sniff" the state file data and see which fields it contains and their types to determine how to handle such state file differences (the "`autoconf` approach" ;)

Clearly, here, there is only 1 possible on-disk format, but I think it would be worth thinking about this and documenting the strategy for how we plan to handle any changes to `vmStateOnDisk`.

Reply:
IMHO adding a version field to the format would be easiest way to handle this kind of issues.